dns.tf

```diff
@@ -35,3 +36,33 @@ resource "aws_route53_record" "ably-zonal" {
   ttl      = var.dns_zonal_record_ttl
   records  = [replace(var.ably_vpc_endpoint_dns_entry, "/^([\\w-]+).(.*)$/", "$1-${each.key}.$2")]
 }
+
+resource "aws_cloudwatch_metric_alarm" "privatelink_disabled_status" {
+  for_each            = toset(var.regions)
+  alarm_name          = "ably-privatelink-region-status-${each.key}"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = "1"
+  metric_name         = "Health"
+  namespace           = "Ably/VPCEndpoint"
+  dimensions = {
+    ServiceName = var.ably_vpc_service_endpoint_name
+  }
+  period            = "60"
+  statistic         = "Maximum"
+  threshold         = "1"
+  alarm_description = "This metric indicates whether an Ably region has been disabled by the Ably Incident Response team"
+}
+
+resource "aws_route53_health_check" "privatelink_disabled_status" {
+  for_each                        = toset(var.regions)
+  type                            = "CLOUDWATCH_METRIC"
+  cloudwatch_alarm_name           = aws_cloudwatch_metric_alarm.privatelink_disabled_status[each.key].alarm_name
+  cloudwatch_alarm_region         = each.key
+  insufficient_data_health_status = "Unhealthy"
```

Comment:
I think if there is insufficient data, we should fail open and let traffic flow rather than disabling the region when perhaps we don't actually want it to be disabled (since e.g. there's a problem with the metric pusher setting the metric)

Reply:
Fixed